Add mgclient

MementoRC opened the pull request:
<!--
Thank you very much for putting in this recipe PR!

This repository is very active, so if you need help with a PR, please let the
right people know. There are language-specific teams for reviewing recipes.

| Language        | Name of review team           |
| --------------- | ----------------------------- |
| c/c++           | `@conda-forge/help-c-cpp`     |
| go              | `@conda-forge/help-go`        |
| java            | `@conda-forge/help-java`      |
| Julia           | `@conda-forge/help-julia`     |
| nodejs          | `@conda-forge/help-nodejs`    |
| perl            | `@conda-forge/help-perl`      |
| python          | `@conda-forge/help-python`    |
| python/c hybrid | `@conda-forge/help-python-c`  |
| r               | `@conda-forge/help-r`         |
| ruby            | `@conda-forge/help-ruby`      |
| rust            | `@conda-forge/help-rust`      |
| other           | `@conda-forge/staged-recipes` |

Once the PR is ready for review, please mention one of the teams above in a
new comment. i.e. `@conda-forge/help-some-language, ready for review!`
Then, a bot will label the PR as 'review-requested'.

Due to GitHub limitations, first time contributors to conda-forge are unable
to ping conda-forge teams directly, but you can [ask a bot to ping the team][1]
using a special command in a comment on the PR to get the attention of the
`staged-recipes` team. You can also consider asking on our [Zulip chat][2]
if your recipe isn't reviewed promptly.

[1]: https://conda-forge.org/docs/maintainer/infrastructure.html#conda-forge-admin-please-ping-team
[2]: https://conda-forge.zulipchat.com

All apologies in advance if your recipe PR does not receive prompt attention.
This is a high volume repository and the reviewers are volunteers. Review times
vary depending on the number of reviewers on a given language team and may be days
or weeks. We are always looking for more staged-recipe reviewers. If you are
interested in volunteering, please contact a member of @conda-forge/core.
We'd love to have the help!
-->

Checklist

- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

MementoRC committed:
Add mgclient and pymgclient recipes

   This PR adds two new packages to conda-forge:

   - **mgclient**: Memgraph C/C++ client library
   - **pymgclient**: Python driver for Memgraph database

MementoRC committed:
(fix) same config?

MementoRC committed:
Add mgclient

MementoRC committed:
(fix) use variants

MementoRC committed:
(fix) use config

MementoRC committed:
(fix) typo in config

MementoRC committed:
Merge branch 'main' into mgclient

MementoRC commented:
@conda-forge/help-c-cpp Hello, ready for review

MementoRC committed:
Merge branch 'main' into mgclient

traversaro commented on recipes/mgclient/build.bat:
```diff
@@ -0,0 +1,30 @@
+set MSYS2_ARG_CONV_EXCL="*"
+set CHERE_INVOKING=1
+bash -lce "./conda_build.sh"
+if errorlevel 1 exit 1
```
The library seems to support MSVC (see https://github.com/memgraph/mgclient/blob/55cc7f5fa291d649e4965a08751a4e05ed75696a/README.md?plain=1#L124). Is there any reason why can't we just compiled with MSVC and keep the recipe simplr?

MementoRC replied:
No disrespect, if I am to be the long term maintainer, t is much simpler to have just one file related to that OS, not to mention much more pleasant and mindful. Dealing with that OS always generates irritations

But of course, if there are technical downstream considerations or conda-forge users inconveniences that I may have missed, please let me know

traversaro commented:
> @traversaro This recipe is ready

Deeply sorry for the review delay, the recipe seems nice, but I wonder if we can simplify it a bit.

traversaro commented on recipes/mgclient/recipe.yaml:
```diff
@@ -0,0 +1,63 @@
+        - m2-sed
+        - m2-which
+  host:
+    - openssl
```
The package installs a shared library, perhaps we can add an appropriate run_exports ?

MementoRC replied:
Thanks, I always forget :(

MementoRC committed:
Update recipe.yaml with m2 variable and sed dependency

Add conditional m2 variable for package requirements and update sed dependency.

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/mgclient/recipe.yaml```) and found some lint.

Here's what I've got...


For **recipes/mgclient/recipe.yaml**:

 * ❌ You are setting `c_stdlib_version` on osx below the current global baseline in conda-forge (11.0).


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/25965920338](https://github.com/conda-forge/conda-forge-webservices/actions/runs/25965920338). Examine the logs at this URL for more detail.</sub>

MementoRC committed:
Add strict option to package_contents in recipe.yaml

MementoRC committed:
Delete recipes/mgclient/conda_build_config.yaml

MementoRC committed:
Merge branch 'main' into mgclient

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/mgclient/recipe.yaml```) and found it was in an excellent condition.

MementoRC committed:
Update recipe.yaml to modify build and run sections

Removed run_exports from build section and added run section for openssl.

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/mgclient/recipe.yaml```) and found some lint.

Here's what I've got...


For **recipes/mgclient/recipe.yaml**:

 * ❌ pin_compatible should be used instead of pin_subpackage for `libmgclient` because it is not a known output of this recipe: ['mgclient'].


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/25966605563](https://github.com/conda-forge/conda-forge-webservices/actions/runs/25966605563). Examine the logs at this URL for more detail.</sub>

MementoRC committed:
Change subpackage pinning from 'libmgclient' to 'mgclient'

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/mgclient/recipe.yaml```) and found it was in an excellent condition.

MementoRC commented:
@traversaro Added the run_exports and made a few simplifications, though holding back on the 'only maintain Unix-style build process' personal preference, if no user implications

traversaro commented:
> if no user implications

It may not be user implications, but it make the recipe much harder to review. I understand preferring not to use autotools clang or other similar things, but in this case using CMake and Visual Studio reduces the build.bat to like ~4 lines. Furthermore, building a project with CMake like this disregard the `%CMAKE_ARGS%`, that is used to easily pass coherent options to all packages built with CMake.

traversaro commented on recipes/mgclient/build.bat:
```diff
@@ -0,0 +1,30 @@
+@echo off
+setlocal enabledelayedexpansion
+
+echo source D:/Miniforge/etc/profile.d/conda.sh       > conda_build.sh
```
This may not be due for local builds or build on self-hosted runners.

MementoRC replied:
Why this should be a consideration?

MementoRC committed:
Refactor build.bat to improve conda activation

Updated build.bat to create conda_build.sh with conditional sourcing of conda.sh based on environment.

MementoRC committed:
Update macOS deployment target in build script

MementoRC committed:
Refactor build.bat for mgclient installation

Removed conda_build.sh wrapper creation and adjusted build process for mgclient.

MementoRC committed:
Clean up build script by removing DLL move logic

Remove unnecessary check for non-Linux and non-macOS platforms regarding DLL installation.

MementoRC committed:
Fix build.bat syntax for Windows compatibility

traversaro commented:
Thanks!
